File: benchmarks/objects_arrays/objects_arrays.rs
struct Person {
    id: i32,
    age: i32,
    weight: f64,
}

impl Person {
    fn new(id: i32, age: i32, weight: f64) -> Self {
        Self { id, age, weight }
    }

    fn age_hundred(&mut self) -> &mut Self {
        while self.age < 100 {
            self.age += 1;
        }
        self
    }

    fn inc_weight(&mut self, increment: f64) -> &mut Self {
        self.weight += increment;
        self
    }

    fn print(&self) -> String {
        format!(
            "Person {} - Age: {}, Weight: {}",
            self.id, self.age, self.weight
        )
    }
}

fn main() {
    let mut people = Vec::new();
    for i in 1..=10000 {
        let person = Person::new(i, 100 - i, 80000.0 - i as f64);
        people.push(person);
    }
    for p in &mut people {
        let id = p.id;
        p.age_hundred().inc_weight(id as f64);
    }
    for p in people {
        println!("{}", p.print());
    }
}

File: tests/benchmarks/objects_arrays/objects_arrays.js
let people = [];

for (let i = 1; i <= 10000; i++) people.push({
    id: i,
    age: 100 - i,
    weight: 80000 - i,

    age_hundred() {
        while (this.age < 100)
            this.age += 1;
        return this;
    },

    inc_weight(increment) {
        this.weight += increment;
        return this;
    },

    print() {
        return `Person ${this.id} - Age: ${this.age}, Weight: ${this.weight}`;
    },
});

for (let p of people) p.age_hundred().inc_weight(p.id);
for (let p of people) console.log(p.print());
